projects/pluto/src/main/java/nl/fairspace/pluto/web/IconsResource.java

```diff
@@ -0,0 +1,55 @@
+package nl.fairspace.pluto.web;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import nl.fairspace.pluto.config.dto.PlutoConfig;
+
+import static nl.fairspace.pluto.config.Urls.ICONS_PATH;
+
+@RestController
+@Slf4j
+@AllArgsConstructor
+public class IconsResource {
+
+    private final PlutoConfig plutoConfig;
+    /**
+     * GET  /api/iconsvg/{icon_name} : returns an icon of specified name if exists.
+     */
+    @GetMapping(value = ICONS_PATH + "{icon_name}", produces = "image/svg+xml")
+    public ResponseEntity<byte[]> iconsvg(@PathVariable String icon_name) {
+        try {
+            InputStream in = getSvgIconInputStream(icon_name);
+            if (in == null || in.available() == 0) {
+                return ResponseEntity.notFound().build();
+            }
+            return ResponseEntity.ok(IOUtils.toByteArray(in));
+        } catch (IOException e) {
+            log.error("Error reading image", e);
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    public String getIconUrl(String name) {
```

Comment:
If to leave it here, I would suggest to make it private as it is for internal usage. But rather I would suggest to make PlutoConfig responsible for the methods and exposing them. Resource/Controller layer should be responsible only for accepting a request and routing it further to dedicated service

Reply:
This one is not only used internally and PlutoConfig is more of a DTO, so I don't think methods belong there. I will see how to organize this better